src/customParser.ts

```diff
@@ -0,0 +1,330 @@
+    if (!startByte) { startByte = 0; }
+    if (!length) { length = compressedBuffer.byteLength - startByte; }
+
+    this.numberRead = 0;
+    this.buffer = compressedBuffer;
+    this.data = new Uint8Array(this.buffer, startByte, length);
+
+    // read first bit
+    this.currentValue = (this.data[0] & (1 << 7)) >> 7;
+    this.nextFibIndex = 1;
+    this.readBitId = 1;
+  }
+
+  nextValue() {
+    this.numberRead++;
+    // read rest of file
+    for (let i = 0; i < 100; i++) {
+      const currentBit8 = this.data[Math.floor(this.readBitId / 8)] & 1 << (7 - (this.readBitId % 8));
+      if (currentBit8 !== 0) {
+        // check if termination bit
+        const previousBit8 = this.data[Math.floor((this.readBitId - 1) / 8)] & (1 << (7 - (this.readBitId - 1) % 8));
+        if ((previousBit8 !== 0) && (this.nextFibIndex !== 0)) {
+          const returnValue = this.currentValue - 1;
+          this.currentValue = 0;
+          this.nextFibIndex = 0;
+          this.readBitId++;
+          return returnValue;
+        } else { // Otherwise update read value
+          this.currentValue += lookup[this.nextFibIndex];
+        }
+      }
+      this.nextFibIndex++;
+      this.readBitId++;
+    }
+    throw Error('THIS IS REALLY BAD');
+  }
+}
+
+class BufferReader {
+  public dataView: DataView;
+  public location: number;
+
+  private arrayBuffer: ArrayBuffer;
+  private flip: boolean;
+
+  constructor(arrayBuffer: ArrayBuffer, flip: boolean) {
+    this.arrayBuffer = arrayBuffer;
+    this.dataView = new DataView(arrayBuffer);
+    this.location = 0;
+    // FLIP ONLY IF READING A NODE FILE
+    this.flip = Boolean(flip);
+  }
+
+  readUint32() {
+    this.location += 4;
+    return this.dataView.getUint32(this.location - 4, this.flip);
+  }
+
+  readUint32Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getUint32(this.location + i * 4, this.flip));
+  }
+  this.location += numberOfValues * 4;
+  return newArray;
+  }
+
+  readUint16Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getUint16(this.location + i * 2, this.flip));
+  }
+  this.location += numberOfValues * 2;
+  return newArray;
+  }
+
+  readUint8Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getUint8(this.location + i));
+  }
+  this.location += numberOfValues;
+  return newArray;
+  }
+
+  readFloat32Array(numberOfValues: number) {
+  const newArray = [];
+  for (let i = 0; i < numberOfValues; i++) {
+    newArray.push(this.dataView.getFloat32(this.location + i * 4, this.flip));
+  }
+  this.location += numberOfValues * 4;
+  return newArray;
+  }
+
+  skip(numberOfBytes: number) {
+    this.location += numberOfBytes;
+  }
+}
+
+function parseCustomFileFormat(asArrayBuffer: ArrayBuffer, flip: boolean) {
```

Comment:
CustomFileFormat is a technically correct name, but we should have a name for the file format maybe? Talk to @nilscognite and @f2cognite about name of file format.

Reply:
I think we've referred to it on the backend as I3DF/Index 3D Format (that's what Nils began calling it).